bdb/temptable.c

```diff
@@ -624,16 +625,16 @@ static struct temp_table *bdb_temp_table_create_main(bdb_state_type *bdb_state,
     tbl->temp_hash_tbl = hash_init_user(hashfunc, hashcmpfunc, 0, 0);
 
     tbl->elements = calloc(tbl->max_mem_entries, sizeof(arr_elem_t));
-    if (tbl->elements == NULL) {
+    if (!tbl->elements) {
+        logmsg(LOGMSG_DEBUG, "%s:%d: failed calloc", __func__, __LINE__);
         bdb_temp_table_close(parent, tbl, bdberr);
-        if (tbl->sql) free(tbl->sql);
+        *bdberr = ENOMEM;
+        if (tbl->sql)
+            free(tbl->sql);
```

Comment:
free(NULL) is a no-op so no need to check for NULL here.

Reply:
I'm only formatting for readability; but i can modify as you suggest.